guides/document_lifecycle.md

```diff
@@ -0,0 +1,269 @@
+  .get(GetParts::IndexId(index, 1))
+  ._source_includes("title")
+  .send()
+  .await?;
+
+let response_body = response.json::<Value>().await?
+println!("{}", response_body["_source"]);
+// OUTPUT: {"year"=>1991}
+```
+
+### Get multiple documents
+
+To get multiple documents, use the `mget` API action:
+
+```rust
+response = client.
+  .mget(MgetParts::Index(index))
+  .body(json!({
+    { "docs": vec![{ "_id": 1 }, { "_id": 2 }] }
+  }))
+  .send()
+  .await?;
+
+let response_body = response.json::<Value>().await?
+for doc in response_body["docs"].iter() {
+  println!("{}", doc["_source"]);
+}
+```
+
+### Check if a document exists
+
+To check if a document exists, use the `exists` API action. The following code checks if the document with ID `1` exists in the `movies` index:
+
+```rust
+client
+  .exists(ExistsParts::IndexId(index, 1))
+  .send()
+  .await?;
+```
+
+### Update a document
+
+To update a document, use the `update` API action. The following code updates the `year` field of the document with ID `1` in the `movies` index:
+
+```rust
+client
+  .update(UpdateParts::IndexId(index, 1))
+  .body(json!({ "doc": { "year": 1995 } }))
+  .send()
+  .await?
+```
+
+Alternatively, you can use the `script` parameter to update a document using a script. The following code increments the `year` field of the of document with ID `1` by 5 using painless script, the default scripting language in OpenSearch:
+
+```rust
+client
+  .update(UpdateParts::IndexId(index, 1))
+  .body(json!({ "script": { "source": "ctx._source.year += 5" }}))
+  .send()
+  .await?;
+```
+
+Note that while both `update` and `index` actions perform updates, they are not the same. The `update` action is a partial update, while the `index` action is a full update. The `update` action only updates the fields that are specified in the request body, while the `index` action overwrites the entire document with the new document.
+
+### Update multiple documents by query
+
+To update documents that match a query, use the `update_by_query` API action. The following code decreases the `year` field of all documents with `year` greater than 2023:
+
+```rust
+client
+  .update_by_query(UpdateByQueryParts::Index(&[index]))
+  .body(json!({
+    "script": { "source": "ctx._source.year -= 1" },
+    "query": { "range": { "year": { "gt": 2023 } }
+  }))
+  .send()
+  .await?;
+```
+
+### Delete a document
+
+To delete a document, use the `delete` API action. The following code deletes the document with ID `1`:
+
+```rust
+client
+  .delete(DeleteParts::IndexId(index, 1))
+  .send()
+  .await?;
+```
+
+By default, the `delete` action is not idempotent. If you try to delete a document that does not exist, or delete the same document twice, you will run into Not Found (404) error. You can make the `delete` action idempotent by setting the `ignore` parameter to `404`:
+
+```rust
+client
+  .delete(DeleteParts::IndexId(index, 1))
+  .params(vec!["ignore", 404])
+  .send()
+  .await?;
+```
```

Comment:
The Rust client doesn't have an exact equivalent of the `ignore` parameter that the Ruby client has added. As the client doesn't directly throw or return errors on a 404, it merely returns the Response regardless of status code, it is left to the user to check the status code themselves.